web/src/upload.ts

```diff
@@ -254,6 +254,59 @@ export function rewire_upload_files(value: typeof upload_files): void {
     upload_files = value;
 }
 
+type PreviousUpload = {
+    size: number | null;
+    metadata: Record<string, string>;
+    creationTime: string;
+    urlStorageKey: string;
+    uploadUrl: string | null;
+    parallelUploadUrls: string[] | null;
+};
```

Comment:
It looks like this type information is not "ours," but derived from uppy somehow?  That might bear a comment

Reply:
Fixed at https://github.com/zulip/zulip/compare/31f0f1174dc8d6c6a512cac585e61a072238147e..f1b41f4859ff4ab18f8b2aa2bc9a02e63a886e89, I was wondering if it would make sense to open an issue in tus-js-client to export this type so we don't need a copy here?

Reply:
Yes, that's very reasonable feedback for upstream.

Reply:
#743